### ISSUE TITLE
Licensing clarification

### ISSUE THREAD

**Issue by kengruven** [sociomantic/libdrizzle-redux#217]
This project is under a BSD license, but beyond that, the license situation is not clear to me.  The top-level COPYING file says:

> Copyright (C) 2012 Brian Aker, Andrew Hutchings
> Copyright (C) 2008 Eric Day

while the docs/license.rst file says:

> Copyright (c) 2012, Drizzle Developer Group

Additionally, each source file seems to have a different copyright in its header comment.  For example, libdrizzle/pack.cc is:

> Copyright (C) 2008-2013 Drizzle Developer Group
> Copyright (C) 2008 Eric Day (eday@oddments.org)

while libdrizzle/poll.cc is:

> Copyright (C) 2013 Drizzle Developer Group
> Copyright (C) 2010 Brian Aker, Trond Norbye

and libdrizzle/structs.h is:

> Copyright (C) 2008-2013 Drizzle Developer Group
> Copyright (C) 2008 Eric Day (eday@oddments.org)
> Copyright (C) 2015 Andreas Bok Andersen (andreas.bok@sociomantic.com)

In all, there are 9 different BSD copyrights in the various source files, and each one says "Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer".

Is it correct, then, that I need to include 11 different BSD licenses, one for each of the various copyrights, in order to distribute a binary containing libdrizzle?

IANAL but perhaps a way could be found to consolidate these.

**Comment by andreas-bok-sociomantic** [comment]
Definitely sounds updating headers to consolidate licensing is needs. Admittedly most files were left untouched after we took over the project from the original maintainers. 

@leandro-lucarella-sociomantic, thoughts on this?

**Comment by leandro-lucarella-sociomantic** [comment]
Not really, I wouldn't touch anything to avoid problems. I guess if you need to redistribute in binary form, just include all the copyrights. It is sort of normal for different files to have different copyrights.

**Comment by andreas-bok-sociomantic** [comment]
However I guess I should change the copyright with my name to: 
```
Copyright (C) [some-year(s)] sociomantic labs GmbH
```

**Comment by leandro-lucarella-sociomantic** [comment]
Add it to the files we touched. Yes.

**Comment by kengruven** [comment]
Could we make COPYING and docs/license.rst have the same contents (or point one to the other), and make its contents the union of all the other licenses/copyrights?

Initially, I had assumed that obeying the top-level COPYING file would be sufficient to legally distribute this library.  (Otherwise, why would that file exist, if it's not accurately describing how I can copy it?)  It's only by accident that I discovered I'd be violating 10+ other licenses.

I bet nobody using this library is giving credit properly, because it's not easy to see what to do.  I'm not suggesting spending any effort on *changing* any licenses.  I'm simply proposing that it's a worthy goal for an open-source project to make it easy for people using the project to do the right thing, e.g., by putting all the license information together in one place.

> It is sort of normal for different files to have different copyrights.

It does happen, but in every case I've seen, the main license file either contains all the licenses, or links to them.  For example, on [plcrashreporter](https://github.com/plausiblelabs/plcrashreporter/blob/master/LICENSE), the license file has the primary license, and then sections below that for:

 - same license but different copyright holders
 - different (but compatible) license

Even though there's 2 licenses and 4 copyrights, it's easy to understand just from looking at the LICENSE file.

**Comment by kengruven** [comment]
Another thing I just noticed: every file is has a "BSD" license, but it's not always the same.  Most have a standard-looking 3-clause BSD license, with the third clause that reads:

> The names of its contributors may not be used to endorse or promote products derived from this software without specific prior written permission.

but in one of them, the third clause was changed (in da84cc11) to read the opposite:

> The names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.

I can't tell if this was intentional or not.

**Comment by leandro-lucarella-sociomantic** [comment]
The issue here is we are not the original authors, we just picked an abandoned library because we needed to use it and we thought it was nice to keep our modifications public in case someone else find them useful.

So taking a look at exactly how many licenses are in there and find a way to put all in one file would be a huge task for us as for any other person. Since we don't distribute this library in binary form, it's very hard to justify this investment for us. I guess the original authors should be contacted, etc.

We are more than happy to accept PRs to improve this situation, but I don't see us working on this any time soon.

**Comment by andreas-bok-sociomantic** [comment]
Will close the issue but it might be reopened if we decide to simplify the licensing.

**Comment by leandro-lucarella-sociomantic** [comment]
Why closing it? It is in fact still an (open) issue and an important one. I would leave it open so is visible to  potential contributors that could fix it.

**Comment by andreas-bok-sociomantic** [comment]
> Why closing it? It is in fact still an (open) issue and an important one. I would leave it open so is visible to potential contributors that could fix it.

Reopening
